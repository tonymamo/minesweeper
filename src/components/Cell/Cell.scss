@import '../../stylesheets/variables.scss';

.cell {
    flex: 1;
    background-color: $brand-primary;
    font-weight: 700;
    color: $gray;
    border: 1px solid white;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    
    &:hover {
        background-color: #6FC2E6;
    }
    
    &:before {
        content: "";
        display: block;
        padding-top: 100%;
        float: left;
    }
}

.cell--opened,
.cell--opened:hover {
    background-color: $gray-lightest;
    cursor: default;
}

.cell__image {
    width: 75%;
}

.cell--mine,
.cell--mine:hover {
    background-color: #dd5544;
}

@media all and (min-width: 30em) {
    .cell {
        border-width: 3px;
    }
}