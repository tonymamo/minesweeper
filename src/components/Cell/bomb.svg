<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 48 48" style="enable-background:new 0 0 48 48;" xml:space="preserve">
	<g>
		<path
			fill="#FFFFFF"
			d="M35,12.2l0.8,0.8l-4.4,4.3l1.2,1.2c0.2,0.2,0.3,0.5,0.3,0.8s-0.1,0.6-0.3,0.8l-1.1,1.2c1.1,1.9,1.6,4,1.6,6.1
		c0,1.7-0.3,3.3-1,4.9c-0.7,1.6-1.6,2.9-2.7,4s-2.5,2-4,2.7c-1.6,0.7-3.2,1-4.9,1c-1.7,0-3.3-0.3-4.9-1c-1.6-0.6-2.9-1.5-4-2.7
		s-2-2.5-2.7-4c-0.7-1.5-1-3.2-1-4.9c0-1.7,0.3-3.3,1-4.9c0.6-1.6,1.5-2.9,2.7-4s2.5-2,4-2.7c1.6-0.6,3.2-1,4.9-1
		c2.2,0,4.2,0.5,6.1,1.6l1.2-1.1c0.2-0.2,0.5-0.3,0.8-0.3s0.6,0.1,0.8,0.3l1.2,1.2L35,12.2z M17.6,17.9c-0.3-0.1-0.6-0.1-0.9,0
		c-1.3,0.5-2.4,1.3-3.4,2.2c-1,1-1.8,2.1-2.3,3.4c-0.1,0.3-0.1,0.6,0,0.9c0.1,0.3,0.3,0.5,0.6,0.6c0.2,0.1,0.3,0.1,0.4,0.1
		c0.5,0,0.9-0.2,1.1-0.7c0.4-1,1-1.9,1.8-2.7c0.8-0.8,1.7-1.4,2.7-1.8c0.3-0.1,0.5-0.4,0.6-0.6c0.1-0.3,0.1-0.6,0-0.9
		C18.1,18.2,17.9,18,17.6,17.9z M34.8,11.3c-0.2,0-0.3-0.1-0.4-0.2l-1.6-1.6c-0.1-0.1-0.2-0.3-0.2-0.4c0-0.2,0.1-0.3,0.2-0.4
		c0.1-0.1,0.3-0.2,0.4-0.2c0.1,0,0.3,0.1,0.4,0.2l1.6,1.6c0.1,0.1,0.2,0.2,0.2,0.4c0,0.2-0.1,0.3-0.2,0.4
		C35,11.3,34.9,11.3,34.8,11.3z M36.6,8.6v1.7c0,0.2-0.1,0.3-0.2,0.4c-0.1,0.1-0.2,0.2-0.4,0.2c-0.2,0-0.3-0.1-0.4-0.2
		c-0.1-0.1-0.2-0.2-0.2-0.4V8.6c0-0.2,0.1-0.3,0.2-0.4C35.7,8.1,35.8,8,36,8c0.2,0,0.3,0.1,0.4,0.2C36.5,8.3,36.6,8.4,36.6,8.6z
		 M37.7,11.2c-0.1,0.1-0.3,0.2-0.4,0.2c-0.1,0-0.3-0.1-0.4-0.2c-0.1-0.1-0.2-0.2-0.2-0.4c0-0.1,0.1-0.3,0.2-0.4l1.6-1.6
		c0.1-0.1,0.3-0.2,0.4-0.2c0.2,0,0.3,0.1,0.4,0.2c0.1,0.1,0.2,0.2,0.2,0.4c0,0.1-0.1,0.3-0.2,0.4L37.7,11.2z M38.8,15.4
		c-0.1,0-0.3-0.1-0.4-0.2l-1.6-1.6c-0.1-0.1-0.2-0.2-0.2-0.4c0-0.2,0.1-0.3,0.2-0.4c0.1-0.1,0.2-0.2,0.4-0.2c0.1,0,0.3,0.1,0.4,0.2
		l1.6,1.6c0.1,0.1,0.2,0.3,0.2,0.4c0,0.1-0.1,0.3-0.2,0.4C39.1,15.4,39,15.4,38.8,15.4z M40,12c0,0.2-0.1,0.3-0.2,0.4
		c-0.1,0.1-0.2,0.2-0.4,0.2h-1.7c-0.2,0-0.3-0.1-0.4-0.2c-0.1-0.1-0.2-0.2-0.2-0.4c0-0.2,0.1-0.3,0.2-0.4c0.1-0.1,0.2-0.2,0.4-0.2
		h1.7c0.2,0,0.3,0.1,0.4,0.2C39.9,11.7,40,11.8,40,12z"/>
	</g>
</svg>
