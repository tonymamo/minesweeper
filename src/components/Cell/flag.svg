<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 48 48" style="enable-background:new 0 0 48 48;" xml:space="preserve">
	<g>
		<path
			fill="#FFFFFF"
			d="M12,14.5v22.6c0,0.1-0.1,0.3-0.2,0.4c-0.1,0.1-0.3,0.2-0.4,0.2h-1.2c-0.1,0-0.3-0.1-0.4-0.2
		c-0.1-0.1-0.2-0.3-0.2-0.4V14.5c-0.8-0.4-1.1-1.1-1.1-2c0-0.6,0.2-1.2,0.7-1.6c0.4-0.5,1-0.7,1.6-0.7c0.6,0,1.2,0.2,1.6,0.7
		c0.5,0.4,0.7,1,0.7,1.6C13.1,13.4,12.8,14.1,12,14.5z M39.4,27.3c0,0.3-0.1,0.5-0.2,0.7c-0.1,0.2-0.4,0.3-0.7,0.5
		c-2.6,1.4-4.8,2.1-6.6,2.1c-0.7,0-1.4-0.1-2.2-0.4c-0.8-0.2-1.4-0.5-1.9-0.8c-0.5-0.3-1.2-0.6-2.1-0.8c-0.8-0.3-1.7-0.4-2.6-0.4
		c-2.3,0-5,0.9-8.3,2.6c-0.2,0.1-0.4,0.2-0.6,0.2c-0.3,0-0.6-0.1-0.8-0.3c-0.2-0.2-0.3-0.5-0.3-0.8V16.5c0-0.4,0.2-0.7,0.6-1
		c0.2-0.2,0.7-0.4,1.4-0.8c2.8-1.4,5.3-2.2,7.5-2.2c1.3,0,2.5,0.2,3.6,0.5c1.1,0.4,2.4,0.9,3.9,1.6c0.4,0.2,1,0.3,1.6,0.3
		c0.6,0,1.3-0.1,2.1-0.4s1.4-0.5,2-0.8s1.1-0.6,1.6-0.8c0.5-0.2,0.8-0.4,1-0.4c0.3,0,0.6,0.1,0.8,0.3c0.2,0.2,0.3,0.5,0.3,0.8V27.3z
		"/>
	</g>
</svg>
