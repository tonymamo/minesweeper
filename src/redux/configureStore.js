import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { apiMiddleware } from 'redux-api-middleware';

import rootReducer from './reducers/index';
import logger from './logger';

let middleware = [
    thunk,
    apiMiddleware,
    logger,
];

export const createStoreWithMiddleware = compose(
    applyMiddleware(...middleware),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
)(createStore);

export default function configureStore(initialState) {
    const store = createStoreWithMiddleware(rootReducer, initialState);

    if ( module.hot ) {
        module.hot.accept('./reducers', () => {
            const nextReducer = require('./reducers').default;
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}
