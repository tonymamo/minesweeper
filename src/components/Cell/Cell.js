import React, {Component} from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import bomb from './bomb.svg';
import flag from './flag.svg';
import './Cell.css';

class Cell extends Component {
    render() {
        const {flagged, opened, hasMine, gameOver, adjacentCellMineCount} = this.props;

        const className = classNames({
            'cell': true,
            'cell--opened': opened,
            'cell--mine': opened && hasMine
        });

        return (
            <div className={className} onClick={this.props.onClick} onContextMenu={this.props.onFlag}>
                {
                    !opened && flagged && !gameOver &&
                    <img src={flag} alt="flag" className="cell__image"/>
                }

                {
                    opened && !hasMine && adjacentCellMineCount > 0 &&
                    <span>{adjacentCellMineCount}</span>
                }

                {
                    ((hasMine && opened) || (hasMine && gameOver)) &&
                    <img src={bomb} alt="bomb" className="cell__image"/>
                }
            </div>
        );
    }
}

Cell.defaultProps = {
    flagged: false,
    hasMine: false,
    opened: false,
    adjacentCellMineCount: 0,
    onFlag() {}
};

Cell.propTypes = {
    flagged: PropTypes.bool,
    hasMine: PropTypes.bool,
    opened: PropTypes.bool,
    adjacentCellMineCount: PropTypes.number,
    onFlag: PropTypes.func
};

export default Cell;
