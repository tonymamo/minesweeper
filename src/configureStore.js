import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { apiMiddleware } from 'redux-api-middleware';

import rootReducer from './redux/index';
import logger from './redux/logger';

let middleware = [
    thunk,
    apiMiddleware,
    logger,
];

export const createStoreWithMiddleware = compose(
    applyMiddleware(...middleware),
    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined' ? window.devToolsExtension() : f => f
)(createStore);

export default function configureStore(initialState) {
    const store = createStoreWithMiddleware(rootReducer, initialState);

    if ( module.hot ) {
        module.hot.accept('./redux', () => {
            const nextReducer = require('./redux').default;
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}
