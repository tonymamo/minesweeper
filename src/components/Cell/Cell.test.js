import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import configureStore from '../../redux/configureStore';
import Cell from './Cell';

const store = configureStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><Cell /></Provider>, div);
});

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><Cell /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
