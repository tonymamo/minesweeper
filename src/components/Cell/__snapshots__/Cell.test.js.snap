// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cell"
  onClick={undefined}
  onContextMenu={[Function]}
/>
`;
